ocp_resources/resource.py

```diff
@@ -806,12 +806,12 @@ def create(self, wait: bool = False) -> Optional[ResourceInstance]:
     def delete(self, wait: bool = False, timeout: int = TIMEOUT_4MINUTES, body: Dict[str, Any] | None = None) -> bool:
         self.logger.info(f"Delete {self.kind} {self.name}")
 
-        if self.exists:
-            hashed_data = self.hash_resource_dict(resource_dict=self.instance.to_dict())
-            self.logger.info(f"Deleting {hashed_data}")
-            self.logger.debug(f"\n{yaml.dump(hashed_data)}")
-
         try:
+            if self.exists:
```

Comment:
This code needs to be changed

We should return False if not exists and log a proper error that the resource do not exist.
We need to call delete only if the resource exists.
There is no race, if you want to make sure the resource deleted pass wait=True.

Reply:
At this moment, resource still exists. It's gone less than a second after this check, so the `instance` call fails.

Reply:
> Do you mean that in line 809 the resource exists but in line 810 it's already gone?
Yes, exactly
Sure, will move it, thanks!

Reply:
A resource can exist in line 809 but be gone in 810, if it was marked for automatic deletion, not as a part of delete() call. Should delete() call support deletion of such resources?

Reply:
delete() should act as best effort, if the resource is there when we called we will try to delete it, if the resource is gone while we try we return True (the user wanted the resource to be delete, it was delete even if it not by us)